src/api-service/__app__/onefuzzlib/pools.py

```diff
@@ -712,14 +712,32 @@ def setup(self) -> None:
                 logging.info("creating scaleset: %s", self.scaleset_id)
         elif vmss.provisioning_state == "Creating":
             logging.info("Waiting on scaleset creation: %s", self.scaleset_id)
-            if vmss.identity and vmss.identity.principal_id:
-                self.client_object_id = vmss.identity.principal_id
+            self.set_identity(vmss)
         else:
             logging.info("scaleset running: %s", self.scaleset_id)
+            self.set_identity(vmss)
             self.state = ScalesetState.running
-            self.client_object_id = vmss.identity.principal_id
         self.save()
 
+    def set_identity(self, vmss: Any) -> None:
```

Comment:
How does this handle pre-existing scalesets during the upgrade case?

Reply:
Existing scalesets are handled in the is_authorized function in agent_authorization.py.
- if the scaleset had a system assigned managed identity it would have been assigned to client_object_id 
- if the scaleset used a user assigned managed identity before this change we fall back to the existing behavior of querying the azure.

Unfortunately we cannot update the object id of the scaleset because the token does not contain the scaleset id.

Reply:
If a user starts the creation of a scaleset and another user starts an upgrade shortly afterwards, this could leave that newly created scaleset in a bad place such that it does not have the client_object_id

Reply:
I Updated the logic to make sure we only set the node to running if we are able to set the object_client_id. your use case will result in a failed node creation.